threedigrid_builder/grid/lines_1d2d.py

```diff
@@ -279,6 +282,24 @@ def assign_2d_node(self, cell_tree: shapely.STRtree) -> None:
         self.line[line_idx, 0] = cell_idx
         self.line_coords[:] = np.nan
 
+    def remove_unassigned(self, nodes) -> None:
+        """Removes 1D-2D lines where any of the required nodes is set to null, represented as -9999
+        This is the case when the nodes are outside the 2D domain.
+        """
+        rows_removed = self[np.where(self.line[:, 0] == -9999)]
+        nodes_removed = nodes.id_to_index(rows_removed.line[:, 1])
+        nodes_removed_formatted = nodes.format_message(nodes_removed)
+
+        logger.warning(
+            f"Removing {len(rows_removed)} 1D-2D lines attached to {nodes_removed_formatted} because they are outside of the DEM."
+        )
+
+        new_array = self[np.where(self.line[:, 0] != -9999)]
+        # overwrite all columns of self with the columns of the array where rows with null nodes are deleted
```

Comment:
This should be done already by the `Array.__getitem__` (the `self[]` done on line 297)!

Reply:
discussed: make it `return self[...]`